File: src/ruby/NinjaSurvivalRate/features/step_definitions/ninja_steps.rb
# encoding: utf-8
require 'spec/expectations' 
require 'cucumber/formatter/unicode'

$:.unshift(File.dirname(__FILE__) + '/../../src')
require 'ninja'

#START:belt
Given /^the ninja has a ([a-z]*) level black\-belt$/ do |belt_level|
	@ninja=Ninja.new :belt_level => belt_level 
end
#END:belt

#START:attacked
When /^attacked by [a\s]*(.*)$/ do |opponent|
	@actions=@ninja.attacked_by(opponent)
end
#END:attacked

#START:actions
Then /^the ninja should (.*)$/ do |expected_action|
	@actions.should include expected_action
end
#END:actions

#START:singletable
Given /^a ninja with the following experience$/ do |table|
   @ninja=Ninja.new table.hashes.first
end
#END:singletable

File: src/dotnet/NinjaSurvivalRate/src/NinjaSteps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Cuke4Nuke.Framework;
using NUnit.Framework;
namespace NinjaSurvivalRate
{
  public class NinjaSteps
  	{
    Ninja ninja;
    List<String> actions;
    //START:belt
    [Given(@"^the ninja has a ([a-z]*) level black-belt$")]
    public void TheNinjaHasABlackBelt(String level)
    {
      ninja = new Ninja(level);
    }
    //END:belt
    //START:attacked
    [When(@"^attacked by [a\s]*(.*)$")]
    public void AttackedBy(String opponent)
    {
      actions = ninja.AttackedBy(opponent);
    }
    //END:attacked
    //START:actions
    [Then("^the ninja should (.*)$")]
    public void TheNinjaShould(String action)
    {
      Assert.IsTrue(actions.Contains(action));
    }
    //END:actions
    //START:singletable
    [Given("^a ninja with the following experience$")]
    public void NinjaWithExperience(Table table)
    {
      List<Dictionary<string, string>> hashes = table.Hashes();
      Dictionary<String, String> ninjaProperties = hashes[0];
      ninja = new Ninja(ninjaProperties["belt_level"]);
    }
    //END:singletable
  }
}
